File: WHVM_MVC/Models/Source.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace WHVM_MVC.Models
{
    public partial class Source
    {
        public int SourceId { get; set; }
        public string SourceName { get; set; }
        public DateTime? SourceDateCreated { get; set; }
        public DateTime? SourceDateImported { get; set; }
        public int? SourceFormatId { get; set; }
        public string SourceFilePath { get; set; }
        public SourceFormat SourceFormat { get; set; }
        public ICollection<Clip> Clips { get; set; }
    }
}

File: WHVM.Web/ClientApp/src/classes/Source.ts
import { ISourceFormat } from '../interfaces/ISourceFormat';

export class Source {
    files: [];
    sourceDateCreated: Date;
    sourceDateEnd: string;
    sourceDateImported: string;
    sourceDateStart: string;
    sourceFilePath: string;
    sourceFormatId: number;
    sourceId: number;
    sourceName: string;
    sourceFormat: ISourceFormat;

    constructor() {
    }
}
